src/focus.service.ts

```diff
@@ -509,6 +509,17 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = true;
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
+        visible = false;
+        break;
+      }
+    }
+
+    if (!visible) {
+      return false;
+    }
```

Comment:
This isn't a very solid implementation, it will only work if the element happens to be hidden using the `hidden` class, not if it's hidden through any other means at all. You might look at jquery's implementation of its [`:visible` selector](https://github.com/jquery/jquery/blob/2d4f53416e5f74fa98e0c1d66b6f3c285a12f0ce/src/css/hiddenVisibleSelectors.js#L12) instead, and _only_ run that check if you've identified an element to be the one you "want" to move to, as it is relatively expensive.